### PR TITLE
feat(Filter): Add Filter Component

### LINKED ISSUE #106: Component - Filter
Provide implementation for the filter as documented here:
http://www.patternfly.org/pattern-library/forms-and-controls/filter/

Note : Ang PF implementation can be found here for reference: http://www.patternfly.org/angular-patternfly/#/api/patternfly.filters.component:pfFilter

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**What**:
Adding basic Filter component

**Link to Storybook**:
https://jeff-phillips-18.github.io/patternfly-react/?selectedKind=Filter&selectedStory=Filter

closes #106 

FYI @serenamarie125 @ohadlevy

### CODE REVIEW

File: src/components/Filter/Filter.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import cx from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Filter extends React.Component {
```
Comment (priley86) [inline src/components/Filter/Filter.js]: It's stateless now 😄 ...so I think we can prefer that syntax:
https://hackernoon.com/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc

ex:
https://github.com/patternfly/patternfly-react/blob/master/src/components/Wizard/WizardSidebar.js

Comment (jeff-phillips-18) [reply src/components/Filter/Filter.js]: Updated. Thanks.

File: src/components/Filter/FilterValueSelector.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    if (placeholder || (filterValues && filterValues.length > 1)) {
+      let title;
+      if (currentValue) {
+        title = currentValue.title || currentValue;
+      } else {
+        title = placeholder || filterValues[0].title || filterValues[0];
+      }
+
+      let menuId = 'filterCategoryMenu';
+      menuId += id ? '_' + id : '';
```
Comment (cdcabrera) [inline src/components/Filter/FilterValueSelector.js]: Maybe a template literal with backticks... think there are a couple of these
```
menuId += id ? `_${id}` : ...
```

**Comment by priley86** [comment]
@jeff-phillips-18 much closer here... 

Here is a few suggested changes: https://github.com/priley86/patternfly-react/commit/4d9ab633f2354d0111f9a0908e247eb55bb5b4ca

I am not positive, but you may be able to convert them all to stateless if just using conditional rendering inline vs event handlers on a class (seems it works OK but please test).

Also... it nice to have the added intellisense for sub-classed components (added example there).

Nice job!

File: src/components/Filter/__mocks__/mockFilterExample.js [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+  Grid,
+  Col,
+  Row,
+  Filter,
+  FilterTypeSelector,
+  FilterValueSelector,
+  FilterCategorySelector,
+  FilterCategoryValueSelector
+} from '../../../index';
+import { boolean } from '@storybook/addon-knobs/dist/index';
```
Comment (priley86) [inline src/components/Filter/__mocks__/mockFilterExample.js]: This is throwing an error for me in the console...

Suggestion is to move the boolean up to the `Filter.stories.js` file and pass it down as a prop to the `MockFilterExample`. I tend to define all the Knobs on the `*.stories.js` file, but that is just preference...makes it easier to copy/paste those imports...

File: src/components/Filter/FilterValueSelector.js (hunk 2 above)
Comment (jeff-phillips-18) [reply src/components/Filter/FilterValueSelector.js]: Fixed. Thanks.

File: src/components/Filter/__mocks__/mockFilterExample.js (hunk 3 above)
Comment (jeff-phillips-18) [reply src/components/Filter/__mocks__/mockFilterExample.js]: I've removed the need for the import.

**Comment by jeff-phillips-18** [comment]
Updated to stateless and added intellisense for all sub-classed components

**Comment by jeff-phillips-18** [comment]
Simplified storybook stories.

**Comment by priley86** [comment]
this looks to good me. I am assuming the next step is to test integrating this w/ data tables/list view/etc. For data tables, we should be able to search the rows in the resolved rows/recompose methods (just how we did sort/paging). Adding the filters should work but will have to test it...

**Comment by jeff-phillips-18** [comment]
Yeah, this will need to be integrated into the views by the application. We can add controlled versions of list view, data tables, etc as we see fit.

**Comment by jeff-phillips-18** [comment]
The less/sass files here will not be needed as these selectors have been added to PF core. I will update the PF version and test, then update this PR.

**Comment by jeff-phillips-18** [comment]
Updated to patternfly 3.35.0 and removed filter specific less/sass

**Comment by jeff-phillips-18** [comment]
@cdcabrera correct, thanks. I've removed it.

**Comment by priley86** [comment]
yep aside from the `input`s I'm fine with this and would like to test integration. We can rebase those with `<FormControl type="text"` later on after #131 is in.

good to merge here 😉

**Pull request merged**